### PR TITLE
Final for distutils constants

### PR DESCRIPTION
**Pull request by max-muoto**
This PR starts to use `Final` for the constants within `disutils`.

### CODE REVIEW

File: stdlib/distutils/command/install.pyi [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-from typing import Any, ClassVar
+from typing import Any, ClassVar, Final
 
 from ..cmd import Command
 
-HAS_USER_SITE: bool
-SCHEME_KEYS: tuple[str, ...]
-INSTALL_SCHEMES: dict[str, dict[Any, Any]]
+HAS_USER_SITE: Final[bool]
+SCHEME_KEYS: Final[tuple[str, ...]]
```
Comment (Avasam) [inline stdlib/distutils/command/install.pyi:7]:
```suggestion
SCHEME_KEYS: Final = ('purelib', 'platlib', 'headers', 'scripts', 'data')
```

File: stdlib/distutils/command/install_lib.pyi [hunk 2]
```diff
@@ -1,8 +1,8 @@
-from typing import Any, ClassVar
+from typing import Any, ClassVar, Final
 
 from ..cmd import Command
 
-PYTHON_SOURCE_EXTENSION: str
+PYTHON_SOURCE_EXTENSION: Final[str]
```
Comment (Avasam) [inline stdlib/distutils/command/install_lib.pyi:5]:
```suggestion
PYTHON_SOURCE_EXTENSION: Final = ".py"
```

File: stdlib/distutils/cygwinccompiler.pyi [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
 
 class CygwinCCompiler(UnixCCompiler): ...
 class Mingw32CCompiler(CygwinCCompiler): ...
 
-CONFIG_H_OK: str
-CONFIG_H_NOTOK: str
-CONFIG_H_UNCERTAIN: str
+CONFIG_H_OK: Final[str]
+CONFIG_H_NOTOK: Final[str]
+CONFIG_H_UNCERTAIN: Final[str]
```
Comment (Avasam) [inline stdlib/distutils/cygwinccompiler.pyi:13]:
```suggestion
CONFIG_H_OK: Final = "ok"
CONFIG_H_NOTOK: Final = "not ok"
CONFIG_H_UNCERTAIN: Final = "uncertain"
```

File: stdlib/distutils/log.pyi [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
-DEBUG: int
-INFO: int
-WARN: int
-ERROR: int
-FATAL: int
+DEBUG: Final[int]
+INFO: Final[int]
+WARN: Final[int]
+ERROR: Final[int]
+FATAL: Final[int]
```
Comment (Avasam) [inline stdlib/distutils/log.pyi:7]:
```suggestion
DEBUG: Final = 1
INFO: Final = 2
WARN: Final = 3
ERROR: Final = 4
FATAL: Final = 5
```

File: stdlib/distutils/fancy_getopt.pyi [hunk 5]
```diff
@@ -25,7 +25,7 @@ def fancy_getopt(
     options: list[_Option], negative_opt: Mapping[_Option, _Option], object: Any, args: list[str] | None
 ) -> list[str] | _GR: ...
 
-WS_TRANS: dict[int, str]
+WS_TRANS: Final[dict[int, str]]
```
Comment (Avasam) [inline stdlib/distutils/fancy_getopt.pyi:28]: I'm fairly certain the following above can also be made `Final` (I'm not sure they were even meant to be public)

```python
longopt_pat: Final = r"[a-zA-Z](?:[a-zA-Z0-9-]*)"
longopt_re: Final[Pattern[str]]
neg_alias_re: Final[Pattern[str]]
longopt_xlate: Final[dict[int, int]]
```

File: stdlib/distutils/sysconfig.pyi [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
-PREFIX: str
-EXEC_PREFIX: str
-BASE_PREFIX: str
-BASE_EXEC_PREFIX: str
+PREFIX: Final[str]
+EXEC_PREFIX: Final[str]
+BASE_PREFIX: Final[str]
+BASE_EXEC_PREFIX: Final[str]
 project_base: str
 python_build: bool
```
Comment (Avasam) [inline stdlib/distutils/sysconfig.pyi:12]: I'm fairly certain these can be Final as well (I'm not sure they were even meant to be public)
```suggestion
project_base: Final[str]
python_build: Final[bool]
```

**Review by Avasam — CHANGES_REQUESTED** [review]
I found a few short literal constants that should be consistant across environments that we can specify explicitly (which makes them both Final and Literal).

Other than that small improvement it looks good to me.

If you'd like, could you reflect your changes in the setuptools stubs ? They are intrinsically linked and often improvements in one stub means improvements in the other.
If not, I'll open a PR for it.

**Comment by max-muoto** [comment]
@Avasam Appreciate the comprehensive review, comments addressed here: https://github.com/python/typeshed/pull/12454/commits/3bc7f6039d7ee3093a286be8da330550753c4de4

**Review by srittau — APPROVED** [review]

**Review by Avasam — APPROVED** [review]

File: stdlib/distutils/command/install.pyi [hunk 7]
```diff
@@ -4,8 +4,8 @@ from typing import Any, ClassVar, Final
 
 from ..cmd import Command
 
-HAS_USER_SITE: bool
-SCHEME_KEYS: tuple[str, ...]
+HAS_USER_SITE: Final[bool]
+SCHEME_KEYS: Final = ("purelib", "platlib", "headers", "scripts", "data")
```
Comment (Avasam) [inline stdlib/distutils/command/install.pyi:8]: Well looks like pytype doesn't like this syntax (and I don't see an issue open for it https://github.com/google/pytype/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen+final+tuple )
```suggestion
SCHEME_KEYS: Final[tuple["purelib", "platlib", "headers", "scripts", "data"]]
```

File: stdlib/distutils/command/install.pyi [hunk 8]
```diff
@@ -4,8 +4,8 @@ from typing import Any, ClassVar, Final
 
 from ..cmd import Command
 
-HAS_USER_SITE: bool
-SCHEME_KEYS: tuple[str, ...]
+HAS_USER_SITE: Final[bool]
+SCHEME_KEYS: Final[tuple[purelib, platlib, headers, scripts, data]]
```
Comment (Avasam) [inline stdlib/distutils/command/install.pyi:8]: My bad, these should be `Literal[]`, that's what I get for writing the suggestion from memory.

 ```suggestion
from typing import Literal
SCHEME_KEYS: Final[tuple[Literal["purelib"], Literal["platlib"], Literal["headers"], Literal["scripts"], Literal["data"]]]
```

**Comment by Avasam** [comment]
Thanks !

**Pull request merged**